### PR TITLE
Conn 2311

### PR DESCRIPTION
**Pull request by ptambellini01**

### CODE REVIEW

**Review by TabassumJafri — APPROVED** [review]

**Review by trantang — APPROVED** [review]

File: Product/Production/Services/DocumentDataSubmissionCore/src/main/java/gov/hhs/fha/nhinc/docdatasubmission/outbound/PassthroughOutboundDocDataSubmission.java [hunk 1]
```diff
@@ -91,8 +90,8 @@ protected SubjectHelper getSubjectHelper() {
         return new SubjectHelper();
     }
 
-    protected MessageGeneratorUtilsDocData getMessageGeneratorUtilsDocData() {
-        return MessageGeneratorUtilsDocData.getInstance();
+    protected MessageGeneratorUtils getMessageGeneratorUtils() {
```
Comment (mhpnguyen) [inline Product/Production/Services/DocumentDataSubmissionCore/src/main/java/gov/hhs/fha/nhinc/docdatasubmission/outbound/PassthroughOutboundDocDataSubmission.java]: I don't think you use or utilize this method at all

File: Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/entity/proxy/AbstractEntityDocRetrieveProxy.java [hunk 2]
```diff
@@ -24,31 +24,35 @@
...
-        if (arguments == null || arguments[0] == null) {
-            return new Object[] {};
+    private static final Logger LOG = LoggerFactory.getLogger(AbstractEntityDocRetrieveProxy.class);
+
+    protected String getUrl(String serviceName) {
+        String result = "";
+        try {
+            result = getWebServiceProxyHelper().getUrlLocalHomeCommunity(serviceName);
+        } catch (Exception ex) {
+            LOG.warn("Unable to retreive url for service: " + serviceName);
```
Comment (mhpnguyen) [inline Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/entity/proxy/AbstractEntityDocRetrieveProxy.java]: Why we have extra log in here?  Can it combine into 1

File: Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/entity/proxy/EntityDocRetrieveProxyWebServiceSecuredImpl.java [hunk 3]
```diff
@@ -77,30 +76,14 @@ public RetrieveDocumentSetResponseType respondingGatewayCrossGatewayRetrieve(Ret
                 response = (RetrieveDocumentSetResponseType) client.invokePort(EntityDocRetrieveSecuredPortType.class, "respondingGatewayCrossGatewayRetrieve", message.getRetrieveDocumentSetRequest());
 
             } else {
-                throw new IllegalArgumentException("Could not determine URL for Entity Doc Query endpoint");
+                throw new IllegalArgumentException("Could not determine URL for Entity Doc Query endpoint Secured");
             }
         } catch (Exception ex) {
-            throw new ErrorEventException(ex, "Unable to call Entity Doc Retrieve");
+            throw new ErrorEventException(ex, "Unable to call Entity Doc Retrieve Secured");
```
Comment (mhpnguyen) [inline Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/entity/proxy/EntityDocRetrieveProxyWebServiceSecuredImpl.java:82]: I wonder if response is needed when exception is thrown

**Review by mhpnguyen — CHANGES_REQUESTED** [review]

File: Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/entity/proxy/EntityDocRetrieveProxyWebServiceSecuredImpl.java (hunk 3 above)
Comment (plobre) [reply Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/entity/proxy/EntityDocRetrieveProxyWebServiceSecuredImpl.java:82]: When I converted these exceptions for Failure Logging, I would have added that as an override. Given that it wasn't already done tells me that there was not an explicit response being returned before this change was implemented. I would then say: No, it is not needed.

File: Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/entity/proxy/AbstractEntityDocRetrieveProxy.java [hunk 4]
```diff
@@ -45,8 +45,7 @@ protected String getUrl(String serviceName) {
         try {
             result = getWebServiceProxyHelper().getUrlLocalHomeCommunity(serviceName);
         } catch (Exception ex) {
-            LOG.warn("Unable to retreive url for service: " + serviceName);
-            LOG.warn("Error: " + ex.getMessage(), ex);
+            LOG.warn("Unable to retreive url for service: " + serviceName + " Error: " + ex.getMessage(), ex);
```
Comment (mhpnguyen) [inline Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/entity/proxy/AbstractEntityDocRetrieveProxy.java]: Please use parameterize logging such as LOG.warm ("Unable to retrieve {} ",servcieName

**Review by mhpnguyen — CHANGES_REQUESTED** [review]

File: Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/entity/proxy/AbstractEntityDocRetrieveProxy.java (hunk 4 above)
Comment (ptambellini01) [reply Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/entity/proxy/AbstractEntityDocRetrieveProxy.java]: I'll let it go this time.

**Review by mhpnguyen — APPROVED** [review]

**Pull request merged**